### PR TITLE
BTRX 662 - Project Notifications

### PR DESCRIPTION
**Pull request by vvicario**
## Testing

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: src/main/webapp/project/NewProjectDocuments.js [hunk 1]
```diff
@@ -104,15 +104,17 @@ export const NewProjectDocuments = hh(class NewProjectDocuments extends Componen
...
 
       WizardStep({
         title: this.props.title, step: this.props.step, currentStep: this.props.currentStep,
-        errorMessage: !this.props.errors.attestation && this.props.generalError ? 'Please check previous steps' : 'Please complete all required fields',
-        error: this.props.generalError || this.props.errors.attestation
+        errorMessage: !this.props.errors.attestation && this.props.generalError ? 
+        'Please check previous steps' : !this.props.submitError ? 
+        'Please complete all required fields' : 
+        'Something went wrong. Please try again.',
+        error: this.props.generalError || this.props.errors.attestation || this.props.submitError
```
Comment (rushtong) [inline src/main/webapp/project/NewProjectDocuments.js]: Nit: the logic here, along with everything aligned at the same level, is very hard to follow. I suggest breaking this down into some intermediate variables that might help clarify what the conditions are.

**Comment by vvicario** [comment]
@rushtong I worked on your feedback!

File: src/main/webapp/project/NewProjectDocuments.js [hunk 2]
```diff
@@ -104,15 +104,21 @@ export const NewProjectDocuments = hh(class NewProjectDocuments extends Componen
...
     let errors = false;
-
+    let errorMessage = ''
+    if(!this.props.errors.attestation && this.props.generalError) {
+      errorMessage = 'Please check previous steps';
+    } else if (this.props.submitError) {
+      errorMessage =  'Something went wrong. Please try again.';
+    } else {
+      errorMessage = 'Please complete all required fields';
+    }
```
Comment (rushtong) [inline src/main/webapp/project/NewProjectDocuments.js]: Thank you for this - it's a lot more clear this way.

File: src/main/webapp/project/NewProjectDocuments.js [hunk 3]
```diff
@@ -104,15 +104,21 @@ export const NewProjectDocuments = hh(class NewProjectDocuments extends Componen
...
-
     let errors = false;
-
+    let errorMessage = ''
+    if(!this.props.errors.attestation && this.props.generalError) {
+      errorMessage = 'Please check previous steps';
+    } else if (this.props.submitError) {
+      errorMessage =  'Something went wrong. Please try again.';
+    } else {
+      errorMessage = 'Please complete all required fields';
```
Comment (rushtong) [inline src/main/webapp/project/NewProjectDocuments.js]: See lines 112 and 114 here - are they swapped? It seems to be the last `else` condition would have the `'Something went wrong. Please try again.'` message.

File: src/main/webapp/project/NewProject.js [hunk 4]
```diff
@@ -104,17 +103,25 @@ class NewProject extends Component {
...
+        this.props.createProjectURL, 
+        this.getProject(),
+        this.state.files,
+        this.state.user.displayName,
+        this.state.user.userName
+        ).then(resp => {
+          Project.getProjectType(this.props.serverURL, resp.data.message.projectKey).
+          then(projectType => {
+            window.location.href = [this.props.serverURL, projectType, "show", resp.data.message.projectKey, "?tab=review&new"].join("/");
+          })
```
[the remark this answers is not in the document]
Comment (vvicario) [reply src/main/webapp/project/NewProject.js]: Agree. we have the same issue on consent group. I'll create a technical debt ticket to handle this.

File: src/main/webapp/project/NewProjectDocuments.js (hunk 3 above)
Comment (vvicario) [reply src/main/webapp/project/NewProjectDocuments.js]: They are not swapped, If we have an error after we try to submit the changes then we have to show 'Something went wrong. Please try again.'.
 If not, we have validations error, then we show 'Please complete all required fields'

**Comment by vvicario** [comment]
@rushtong Ready to review!
